### PR TITLE
Loose Catbeast Miniantag

### PR DESCRIPTION
**Pull request by Kurfursten**
* Costs no threat
* Regenerates 1 threat for the round every 20 seconds it is alive (up to 15 at 5 minutes) on the main station.
* Can also regenerate 1 threat by entering a room on the main station it hasn't entered before.
* If threat has reached the threat cap, it begins increasing the threat cap as well (same rate).
* Will not appear if threat is over 50
* Spawns with a random humiliating jumpsuit (darkholme, maid, schoolgirl, psychedelic, clown, xenoarcheologist, etc.)
* Also gets a messenger bag, sandals
* After 1.5 minutes if it is alive a command report will announce its presence.
* Only one catbeast can appear per shift (subject to change)
* Scorescreen displays how many seconds lived, how many rooms entered, and total threat generated.
* Objective 1: Survive 5 minutes on the main station. Objective 2: Enter 30 rooms on the main station.

🆑 
* rscadd: Mangy catbeasts can now be generated by dynamic! Kill them before they attract other dangerous antagonists!

### CODE REVIEW

**Comment by ShiftyRail** [comment]
Would need a bit of wrapping to be more presentable
Make a command report or something

**Comment by Irradiation** [comment]
Make the catbeast black so we can go full-on "black cats bring bad luck".

**Comment by Kurfursten** [comment]
@ShiftyRail Maybe on a 5 minute delay (if still alive of course). And yeah, we can do a robe or something (just until they inevitably start looting to hide their tail).

@Irradiation I'll do it if you code black catbeasts to begin with and then I just rebase and use it.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Seems worrying that the threat generation is fast and uncapped so a single catbeast can bring the fucking apocalypse if they manage to hide in a maint falsewall for 20 minutes

The threat gain should start fast and have diminishing returns

**Comment by Kurfursten** [comment]
It's capped at 100, that's not a bad idea though. For the record, it would take more than half an hour unfound to make that much threat (from 0).

Another option might be, if the catbeast stays in one place too long it sends a fax with the area name. Since we're making regular checkups it should be easy too.

**Comment by SonixApache** [comment]
->spawn in as catbeast
 ->steal large air tank, spacesuit, one set of rods and fuck off to space
 ->build space platform on random Z-level
 ->ghost
 ->laugh as station is enveloped by infinite chaos

that aside, black catbeasts would be a major beneficial addition to the code

**Comment by Kurfursten** [comment]
Also a good catch 👌 

Should be easy enough to ensure they are on the station with similar code that we use for the operative nuke

**Comment by Kurfursten** [comment]
Just had what I think is a brilliant idea:

Time survived might contribute only up to say, 5 minutes (15 threat)
Then further threat is accrued by visiting rooms. There are, if I counted right, 138 distinct room areas on Boxstation. Some of those are near impossible for a catbeast to reach (AI core, 8 xenobiology areas, and secure armory stand out). I count 25ish public rooms. How about 1 threat for each unique station area visited?

File: code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -299,3 +299,20 @@
...
+	required_enemies = list(0,0,0,0,0,0,0,0,0,0)
+	required_candidates = 1
+	weight = 1
+	cost = 0
+	requirements = list(0,0,0,0,0,0,0,0,0,0)
+	logo = "catbeast-logo"
+
+/datum/dynamic_ruleset/midround/from_ghosts/catbeast/acceptable(var/population=0,var/threat=0)
+	if(threat>50) //We're threatening enough!
+		message_admins("Rejected catbeast ruleset, [threat] threat was over 50."
```
Comment (Exxion) [inline code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm]:
```suggestion
		message_admins("Rejected catbeast ruleset, [threat] threat was over 50.")
```

**Comment by Kurfursten** [comment]
Major content update implemented. Everything in the OP described, plus
* Wiki
* Mode icon
* Rewrote our desires_role stuff in trim_candidates to properly use required_pref (this allows us to have multiple roles like catbeast share ROLE_MINOR preference)

Tested, it's all working

EDIT: Now with less "included test_box"!

**Comment by Arthurdentist** [comment]
Add 'Kingston' as a possible name please

**Comment by Shadowmech88** [comment]
Is the threat generation tied to the antag mind, or to the black catbeast mob?

**Comment by Kurfursten** [comment]
Generation is done via the Role process which checks whether the associated mind's body is alive, on Z_STATION, and not in space.

As of right now no one has coded a black catbeast so I wouldn't hold out for that being a launch feature.

**Comment by Intigracy** [comment]
I'll code black catbeasts

**Comment by Shadowmech88** [comment]
>which checks whether the associated mind's body is alive

I'm not familiar with how minds work, but what if the catbeast is de-brained and put inside a non-catbeast living body? Or what would happen if a brain is periodically re-inserted into the catbeast body with the intention of temporarily pumping the threat a bit?

**Comment by Kurfursten** [comment]
It would only make threat up to 5 minutes since the change

You bring up an interesting point though. I could add a check where if the brain is put into a non catbeast the player loses the role. But on the other hand, mad science experiments involving reviving catbeasts offending the syndicate enough to send more antags... kind of sounds emergent gameplay? Remember this would require their mind, so you would have to get a player totally willing to be tortured through repeated revival for this to work.

**Comment by Kurfursten** [comment]
In light of hearing that Deity's threat increase thing got out of hand quickly, I'm planning to get rid of the mask and reduce the delay before the announcement to a 90 second headstart as a means of giving the crew more opportunity for a quick lynch. Will make updates when Intigracy's black catbeast happens.

**Comment by Kurfursten** [comment]
Black catbeast didn't happen so I'm gonna leave that to Intigracy to add in a later PR. This is good to go.

File: code/datums/gamemode/role/catbeast.dm [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		return FALSE
+	return A
+
+/datum/role/catbeast/proc/increment_threat(var/reason)
+	var/datum/gamemode/dynamic/D = ticker.mode
+	if(!istype(D))
+		return //It's not dynamic!
+	threat_generated++
+	if(D.threat >= D.threat_level)
+		D.threat_level = min(D.threat_level+1,100)
```
Comment (DeityLink) [inline code/datums/gamemode/role/catbeast.dm]: NOOOOOOOOOOOO

**Comment by DeityLink** [comment]
First @ShiftyRail now you, guys stop incrementing threat_level already, it's got almost nothing to do with threat!

**Pull request merged**